### ISSUE TITLE
Add proper diagnostic when an integer is passed to the magic attribute with an unspecified type

### ISSUE THREAD

**Issue by winterqt** [jam1garner/binrw#79]
```rust
#[binrw]
#[br(little, magic = 1)]
struct Header {
    x: u8,
}
```

```rust
error: expected type, found `;`
 --> src/main.rs:5:1
  |
5 | #[binrw]
  | ^^^^^^^^
  | |
  | expected type
  | while parsing this item list starting here
  | the item list ends here
  |
  = note: this error originates in the attribute macro `binrw` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0046]: not all trait items implemented, missing: `MagicType`, `MAGIC`
 --> src/main.rs:5:1
  |
5 | #[binrw]
  | ^^^^^^^^ missing `MagicType`, `MAGIC` in implementation
  |
  = note: this error originates in the attribute macro `binrw` (in Nightly builds, run with -Z macro-backtrace for more info)
  = help: implement the missing item: `type MagicType = Type;`
  = help: implement the missing item: `const MAGIC: <Self as HasMagic>::MagicType = value;`
```

`binrw` version: 0.8.0

**Comment by jam1garner** [comment]
Implementation guide for anyone interested:
1. [Relevant code is in this impl](https://cs.github.com/jam1garner/binrw/blob/16ccad43a51e84896eb7ecea2b51659702f6fae0/binrw_derive/src/parser/types/magic.rs?q=suffix#L77-L102)
2. For literals of type `Lit::Float` or `Lit::Int`, check if `x.suffix().is_empty()`. If `true`, then output a `compile_error!()` with a span matching that of the literal. This can probably be done by just constructing a `syn::Error` in the above function.
3. If for whatever `syn::Error` isn't enough to get the desired error diagnostic, find the right place for using [`to_compile_error`](https://docs.rs/syn/latest/syn/struct.Error.html#method.to_compile_error) to actually emit the error.

**Comment by csnover** [comment]
This was fixed by dd4fa9b4610369f7f9c2034d205563edf468a1f5.
